### PR TITLE
Minor improvements to the interface for sygus sampler

### PR DESCRIPTION
**Pull request by ajreynol**
This is 1/3 PRs for merging the sygus query generator branch https://github.com/ajreynol/CVC4/tree/sygusQueryGen.

### CODE REVIEW

File: src/theory/quantifiers/sygus/cegis.cpp [hunk 1]
```diff
@@ -527,8 +527,9 @@ bool Cegis::sampleAddRefinementLemma(const std::vector<Node>& candidates,
         // mark this as a CEGIS point (no longer sampled)
         d_cegis_sample_refine.insert(i);
         std::vector<Node> vars;
+        d_cegis_sampler.getVariables(vars);
```
Comment (4tXJ7f) [inline src/theory/quantifiers/sygus/cegis.cpp]: It looks like `vars` is not used. Can we remove it?

File: src/theory/datatypes/datatypes_sygus.cpp [hunk 2]
```diff
@@ -895,8 +895,9 @@ Node SygusSymBreakNew::registerSearchValue(
               Trace("sygus-rr-debug")
                   << "; unsound: are not equivalent for : " << std::endl;
               std::vector<Node> vars;
+              its->second.getVariables(vars);
```
Comment (4tXJ7f) [inline src/theory/datatypes/datatypes_sygus.cpp]: It looks like `vars` is only used for the assertion check. Move it into an `if (Configuration::isAssertionBuild())`?

Comment (ajreynol) [reply src/theory/datatypes/datatypes_sygus.cpp]: It's also used in the printing within the for loop: https://github.com/CVC4/CVC4/pull/2326/files#diff-b5f4e26d011c1b54d669904177e990efL903.

File: src/theory/quantifiers/sygus/cegis.cpp (hunk 1 above)
Comment (ajreynol) [reply src/theory/quantifiers/sygus/cegis.cpp]: Yes good catch.

**Comment by ajreynol** [comment]
Anything more for this one?

File: src/theory/datatypes/datatypes_sygus.cpp (hunk 2 above)
Comment (4tXJ7f) [reply src/theory/datatypes/datatypes_sygus.cpp]: Ah yes, but then why not just move that function call into an if statement guarded by `Configuration::isAssertionBuild() || Trace.on("sygus-rr-debug")`? Like that we don't incur the overhead of copying the variables when we are running in production without tracing.

Comment (ajreynol) [reply src/theory/datatypes/datatypes_sygus.cpp]: Actually, this output should be part of the exception on line 921 (I'm surprised its not already). I'll refactor.

**Pull request merged**